backend/battles/views.py

```diff
@@ -39,10 +43,28 @@ def get_context_data(self, **kwargs):
         user = User.objects.get(id=self.request.user.id)
         battle = self.kwargs["pk"]
         if user.teams.filter(battle=battle).exists():
-            context["user_has_team"] = True
-
+            messages.info(
+                self.request,
+                "You've chosen your team for this battle. Check your email for results",
+            )
         return context
 
+    def form_valid(self, form):
+        self.object = form.save()
```

Comment:
I'm not 100% sure but I think you don't need to use self.object. You could just:
```python
team = form.save()
```
It would make the code clearer

Reply:
You could also use "team" in the next lines